tools/c7n_awscc/c7n_awscc/actions.py

```diff
@@ -46,9 +47,19 @@ def process(self, resources):
             )
 
     def get_patch(self, r):
-        tgt = dict(r)
+        # deep copy is expensive, but we need to remove read only and create only properties
+        # to generate a valid patch.
+        current = copy.deepcopy(r)
+
+        # the action's schema reflects updatable properties
+        updatable = {k for k in self.schema["properties"].items() if k != "type"}
+        for k in list(set(current) - updatable):
+            del current[k]
```

Comment:
If we're only excluding or override top-level keys, could we still get the key skipping benefits without deepcopy? Something like:

```suggestion
        # the action's schema reflects updatable properties
        updatable = {k for k in self.schema["properties"].items() if k != "type"}
        current = {k: r[k] for k in updatable}
```

Reply:
yeah.. we might be able to get away with the need for deep copy, since we're only doing top level key replacement here, my concern is if we need to be doing partial updates, see referenced issue `WorkGroupConfiguration` then we may need to be doing patch updates into dicts in which case we do need a deep copy to preserve the original.

Reply:
Ahhh gotcha, okie safety wins. Thanks for the clarification :+1:

Reply:
went ahead and added patch functionality, so it might be worth revisiting this as we're only doing top level key/value replacement, re can get away with shallow copy.